src/shims/time.rs

```diff
@@ -159,4 +160,25 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
         u64::try_from(duration.as_nanos())
             .map_err(|_| err_unsup_format!("programs running longer than 2^64 nanoseconds are not supported").into())
     }
+
+    fn mach_timebase_info(&mut self, info_op: OpTy<'tcx, Tag>) -> InterpResult<'tcx, i32> {
+        let this = self.eval_context_mut();
+
+        this.assert_target_os("macos", "mach_timebase_info");
+        this.check_no_isolation("mach_timebase_info")?;
+
+        let info = this.deref_operand(info_op)?;
+
+        // Since we return nanoseceonds instead of ticks from
+        // `mach_absolute_time`, we don't need to scale the absolute
+        // time.
+        let (numer, denom) = (1,1);
+        let imms = [
+            immty_from_int_checked(numer, this.libc_ty_layout("uint32_t")?)?,
+            immty_from_int_checked(denom, this.libc_ty_layout("uint32_t")?)?
+        ];
+
+        this.write_packed_immediates(info, &imms)?;
+        Ok(0)
```

Comment:
The docs say this returns `KERN_SUCCESS`... do you have a documentation for that being 0?

Reply:
Hmm, I can't actually find actual documentation from Apple about what the value of `KERN_SUCCESS` will be. The most convincing argument I can present is that LLVM also uses `0`: https://github.com/llvm/llvm-project/blob/cdc514e4c67f268b07863bbac3d8d7e0d088186c/lldb/source/Plugins/Process/Utility/RegisterContextDarwinConstants.h

Would it be possible to instead read the #define'd `KERN_SUCCESS` instead? I tried `this.eval_libc("KERN_SUCCESS")?.to_i32()` but the name doesn't resolve possibly because it's not looking in the right header file?

Reply:
> Would it be possible to instead read the #define'd KERN_SUCCESS instead? I tried this.eval_libc("KERN_SUCCESS")?.to_i32() but the name doesn't resolve possibly because it's not looking in the right header file?

I couldn't find `KERN_SUCCESS` in the libc crate so we cannot read it like that.

Reply:
> Hmm, I can't actually find actual documentation from Apple about what the value of KERN_SUCCESS will be. The most convincing argument I can present is that LLVM also uses 0

That's a good argument. Please add a comment here like
```rust
Ok(0) // KERN_SUCCESS
```

Reply:
`KERN_SUCCESS` has been defined as zero in mach/kern_return.h for decades. You'll be fine.

Reply:
@gparker42 thanks :)